tests/components/cookidoo/test_init.py

```diff
@@ -0,0 +1,126 @@
+"""Unit tests for the cookidoo integration."""
+
+from unittest.mock import AsyncMock
+
+from cookidoo_api import CookidooAuthException, CookidooRequestException
+import pytest
+
+from homeassistant.components.cookidoo import async_setup_entry
+from homeassistant.components.cookidoo.const import DOMAIN
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+
+from . import setup_integration
+
+from tests.common import MockConfigEntry
+
+
+@pytest.mark.usefixtures("mock_cookidoo_client")
+async def test_load_unload(
+    hass: HomeAssistant,
+    cookidoo_config_entry: MockConfigEntry,
+) -> None:
+    """Test loading and unloading of the config entry."""
+    await setup_integration(hass, cookidoo_config_entry)
+
+    entries = hass.config_entries.async_entries(DOMAIN)
+    assert len(entries) == 1
+
+    assert cookidoo_config_entry.state is ConfigEntryState.LOADED
+
+    assert await hass.config_entries.async_unload(cookidoo_config_entry.entry_id)
+    assert cookidoo_config_entry.state is ConfigEntryState.NOT_LOADED
+
+
+@pytest.mark.parametrize(
+    ("exception", "status"),
+    [
+        (CookidooRequestException, ConfigEntryState.SETUP_RETRY),
+        (CookidooAuthException, ConfigEntryState.SETUP_ERROR),
+    ],
+)
+async def test_init_failure(
+    hass: HomeAssistant,
+    mock_cookidoo_client: AsyncMock,
+    status: ConfigEntryState,
+    exception: Exception,
+    cookidoo_config_entry: MockConfigEntry,
+) -> None:
+    """Test an initialization error on integration load."""
+    mock_cookidoo_client.login.side_effect = exception
+    await setup_integration(hass, cookidoo_config_entry)
+    assert cookidoo_config_entry.state == status
+
+
+@pytest.mark.parametrize(
+    ("exception", "expected"),
+    [
+        (CookidooRequestException, ConfigEntryNotReady),
+        (CookidooAuthException, ConfigEntryAuthFailed),
+    ],
+)
+async def test_init_exceptions(
+    hass: HomeAssistant,
+    mock_cookidoo_client: AsyncMock,
+    exception: Exception,
+    expected: Exception,
+    cookidoo_config_entry: MockConfigEntry,
+) -> None:
+    """Test an initialization error on integration load."""
+    cookidoo_config_entry.add_to_hass(hass)
+    mock_cookidoo_client.login.side_effect = exception
+
+    with pytest.raises(expected):
```

Comment:
can we use `matches` to match the name of the exception?

Reply:
@joostlek Wym? The pytest.raises already checks the type of the Exception raised, no?

Reply:
Yes, but the name is translated, so we also want to test the content of the exception